### DISCUSSION TITLE
Question

### DISCUSSION THREAD

**Discussion by AntoineD** [pawamoy/dependenpy#59] [General]
I'm looking for a similar tool that would determine the dependencies between classes (to compute coupling / cohesion metrics). Are you aware of such a tool?

**Comment by pawamoy** [comment]
Hi @AntoineD!

I'm not, but I would like to use such a tool indeed. If you were to write it yourself, you'd need a powerful inference engine to know the type of each object passed to each function call. `astroid` could help, or maybe the still-being-built [`libstatic`](https://github.com/tristanlatr/libstatic/) project.

**Reply by AntoineD** [reply]
I've been wondering about writing such a tool, but unfortunately I do not have enough spare time. I was thinking about [jedi](https://jedi.readthedocs.io/en/latest/index.html) for the code analysis.

Thank you anyway!
